File: wallet-app-backend/WalletApp.Application/Common/Transaction/AddUserTransaction/AddUserTransactionCommand.cs
using WalletApp.Application.Abstractions.Messaging;

namespace WalletApp.Application.Common.Transaction.AddUserTransaction;

public sealed record AddUserTransactionCommand(
    Guid UserId,
    string Title,
    decimal Price,
    Guid CurrencyId,
    Guid CategoryId,
    DateTime Date,
    bool IsDefault,
    string? Description,
    string? TextColor,
    string? BackgroundColor
) : ICommand;

File: wallet-app-ui/src/models/apiTypes/transaction/addUserTransaction/addUserTransaction.command.ts
export class AddUserTransactionCommand {
  constructor(
    public title: string,
    public price: number,
    public currencyId: string,
    public categoryId: string,
    public date: Date,
    public isDefault: boolean,
    public isTemplate?: boolean,
    public description?: string,
    public textColor?: string,
    public backgroundColor?: string,
  ) {}
}
